### PR TITLE
credit card fraud code example updated to run smoothly with latest bo…

### PR DESCRIPTION
**Pull request by alifarhid**
This is very high priority as we need to communicate it asap to Saturn Cloud.

### CODE REVIEW

File: ml/credit-card-fraud.ipynb [hunk 1]
```diff
@@ -49,10 +49,89 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 1,
+   "execution_count": 20,
```
Comment (hadia206) [inline ml/credit-card-fraud.ipynb:52]: Could you move this cell and the next one (print ranks) to be before the cell describing the imports? 
Also, add a cell explaining what you're doing in these two cells. You can point to the getting started in [here](https://docs.bodo.ai/latest/source/installation_and_setup/ipyparallel.html#getting-started-on-your-machine)

File: ml/credit-card-fraud.ipynb [hunk 2]
```diff
@@ -49,10 +49,89 @@
...
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "%%px\n",
+    "from mpi4py import MPI\n",
+    "comm = MPI.COMM_WORLD\n",
+    "print(f\"Hello World from rank {comm.Get_rank()}. total ranks={comm.Get_size()}\")"
```
Comment (hadia206) [inline ml/credit-card-fraud.ipynb:118]: You can use Bodo API instead.
```suggestion
    "print(f\"Hello World from rank {bodo.get_rank()}. total ranks={bodo.get_size()}\")"
```

**Review by hadia206 — APPROVED** [review]
looks good to me.
I left minor comments.
Thanks @alifarhid

**Comment by hadia206** [comment]
It looks like you're running on 4 cores only. Please update the instance description where it says `The current results are based on running on one m5.12xlarge instance (24 cores, 192GiB memory)` to reflect your settings.

**Comment by alifarhid** [comment]
Yes @hadia206 I read that and I realized I need to update the documentation at the top. I am working on it.

File: ml/credit-card-fraud.ipynb [hunk 3]
```diff
@@ -49,10 +49,89 @@
...
+     "output_type": "stream",
+     "text": [
+      "Starting 4 engines with <class 'ipyparallel.cluster.launcher.MPIEngineSetLauncher'>\n",
+      "100%|█████████████████████████████████████████| 4/4 [00:06<00:00,  1.69s/engine]\n"
+     ]
+    }
+   ],
+   "source": [
+    "import ipyparallel as ipp\n",
+    "import psutil; n = min(psutil.cpu_count(logical=False), 4)\n",
```
Comment (ehsantn) [inline ml/credit-card-fraud.ipynb:67]: Bodo community edition works on 8 cores. Are we using 4 to limit the amount of output?

File: ml/credit-card-fraud.ipynb (hunk 2 above)
Comment (ehsantn) [reply ml/credit-card-fraud.ipynb:118]: I agree, let's use Bodo APIs instead. Here is the updated docs: https://docs.bodo.ai/latest/source/installation_and_setup/ipyparallel.html#setupverify-multiple-hosts

Comment (alifarhid) [reply ml/credit-card-fraud.ipynb:118]: Sure, I changed to Bodo APIs instead.

File: ml/credit-card-fraud.ipynb (hunk 3 above)
Comment (alifarhid) [reply ml/credit-card-fraud.ipynb:67]: Yes, mainly two reasons why I ran with 4 cores:
1- The outputs were overwhelmingly large as it prints per core. 
2- Some developers do not have a laptop with 8 physical cores. Mostly 4 physical cores with 2 thread per core. Kind of being more inclusive. But people know community edition is 8 cores, so they can run with 8 cores, see the difference and be surprised!

**Pull request merged**